libcontainer/nsenter/cloned_binary.c

```diff
@@ -88,28 +91,61 @@ static void *must_realloc(void *ptr, size_t size)
 static int is_self_cloned(void)
```

Comment:
How about just use `CLONED_BINARY_ENV` to avoid potential loop that we have not found yet?
I don't know whether this is more safety or not!

For example:
```
static int is_self_cloned(void) {
    return getenv(CLONED_BINARY_ENV) && strcmp(getenv(CLONED_BINARY_ENV), "1")==0;
}
```

Reply:
I think there is just one disadvantage, even if we have protected run binary file, we will call ensure_clone_binary first.

Reply:
Because I don't want to be in a situation where someone passes `CLONED_BINARY_ENV` to `runc` and that results in an unsafe setup.

In fact, the *only* reason why `CLONED_BINARY_ENV` exists in this PR is to stop the case where `/usr/bin/runc` is on a read-only mount (or is an unlinked file) and we think that makes the binary safe -- but in those cases it is *not* safe (and we need to apply the protection).

I don't want to provide anyone the ability to opt out of this protection -- if you want to, you can bind-mount a sealed `memfd` over `/usr/bin/runc` if you want to "opt out" but apart from that there is no reason to opt-out now that the primary protection is a read-only bind-mount which takes up basically no memory.

Reply:
For this topic, I have no other opinions.